loader/loader_test.go

```diff
@@ -349,6 +349,87 @@ func TestLoadErrors(t *testing.T) {
 	})
 }
 
+func TestLoadFileURL(t *testing.T) {
+	files := map[string]string{
+		"/a/a/1.json": `1`,        // this will load as a directory (e.g., file://a/a)
+		"b.json":      `{"b": 2}`, // this will load as a normal file
+		"c.json":      `3`,        // this will loas as rooted file
+	}
+	test.WithTempFS(files, func(rootDir string) {
+
+		paths := mustListPaths(rootDir, false)[1:]
+		sort.Strings(paths)
+
+		for i := range paths {
+			paths[i] = "file://" + paths[i]
+		}
+
+		paths[2] = "c:" + paths[2]
+
+		result, err := All(paths)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		exp := parseJSON(`{"a": 1, "b": 2, "c": 3}`)
+		if !reflect.DeepEqual(exp, result.Documents) {
+			t.Fatalf("Expected %v but got %v", exp, result.Documents)
+		}
+	})
+}
+
+func TestUnsupportedURLScheme(t *testing.T) {
+	_, err := All([]string{"http://openpolicyagent.org"})
+	if err == nil || !strings.Contains(err.Error(), "unsupported URL scheme: http://openpolicyagent.org") {
+		t.Fatal(err)
+	}
+}
+
+func TestSplitPrefix(t *testing.T) {
+
+	tests := []struct {
+		input     string
+		wantParts []string
+		wantPath  string
+	}{
+		{
+			input:    "foo/bar",
+			wantPath: "foo/bar",
+		},
+		{
+			input:     "foo:/bar",
+			wantParts: []string{"foo"},
+			wantPath:  "/bar",
+		},
+		{
+			input:     "foo.bar:/baz",
+			wantParts: []string{"foo", "bar"},
+			wantPath:  "/baz",
+		},
+		{
+			input:    "file://a/b/c",
+			wantPath: "file://a/b/c",
```

Comment:
nits:
* It might be worth adding a test case for windows style paths like: `file:///c:/a/b/c`
* I think too that these should be `file:///a/b/c` (3 `/`'s, `file://` + `/a/b/c`) https://play.golang.org/p/2Pej43iwYQb as-is it trims off the `a` which is probably not expected.